dali/operators/image/convolution/laplacian_params.h

```diff
@@ -0,0 +1,293 @@
+        has_scale_const_input_{spec.HasArgument(scaleArgName)},
+        normalize_{spec.GetArgument<bool>(normalizeArgName)} {}
+
+  void ObtainLaplacianArgs(const ArgumentWorkspace &ws, int nsamples) {
+    ObtainWindowSizes(ws, nsamples);
+    ObtainScales(ws, nsamples);
+  }
+
+  inline const std::array<std::array<int, axes>, axes> &GetWindowSizes(int sample_idx) {
+    return window_sizes_[sample_idx];
+  }
+
+  inline const std::array<float, axes> &GetScales(int sample_idx) {
+    return scales_[sample_idx];
+  }
+
+  inline int GetTotalWindowSizes(int sample_idx) {
+    auto &window_sizes = GetWindowSizes(sample_idx);
+    auto acc_sizes = [](int acc, const std::array<int, axes> &win_sizes) {
+      return std::accumulate(win_sizes.begin(), win_sizes.end(), acc);
+    };
+    return std::accumulate(window_sizes.begin(), window_sizes.end(), 0, acc_sizes);
+  }
+
+ private:
+  void ObtainWindowSizes(const ArgumentWorkspace &ws, int nsamples) {
+    if (!has_win_size_t_input_) {
+      int prev_size = window_sizes_.size();
+      if (prev_size < nsamples) {
+        auto window_sizes = prev_size == 0 ? ObtainConstWindowSizes() : window_sizes_[0];
+        window_sizes_.resize(nsamples, window_sizes);
+      }
+    } else {
+      window_sizes_.resize(nsamples);
+      for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+        const auto &tv = ws.ArgumentInput(kWindowSizeArgName);
+        const auto &tensor = tv[sample_idx];
+        const auto &shape = tensor.shape();
+        auto vol = volume(shape);
+        DALI_ENFORCE(vol == 1 || vol == axes || vol == num_windows,
+                     make_string("Argument `", kWindowSizeArgName, "` for sample ", sample_idx,
+                                 " is expected to have 1, ", axes, " or ", axes, "x", axes,
+                                 " elements, got ", vol, "."));
+        SetSampleWindowSizes(window_sizes_[sample_idx],
+                             make_span(tensor.template data<int>(), vol));
+        VerifyWindowSizes(window_sizes_[sample_idx], sample_idx);
+      }
+    }
+  }
+
+  std::array<std::array<int, axes>, axes> ObtainConstWindowSizes() {
+    std::vector<int> tmp;
+    if (!spec_.TryGetRepeatedArgument<int>(tmp, kWindowSizeArgName)) {
+      int scalar = spec_.GetArgument<int>(kWindowSizeArgName);
+      tmp.assign(axes, scalar);
+    }
+    std::array<std::array<int, axes>, axes> window_sizes;
+    auto in_size = tmp.size();
+    DALI_ENFORCE(in_size == 1 || in_size == axes || in_size == num_windows,
+                 make_string("Argument `", kWindowSizeArgName, "` is expected to have 1, ", axes,
+                             " or ", axes, "x", axes, " elements, got ", in_size, "."));
+    SetSampleWindowSizes(window_sizes, make_span(tmp));
+    VerifyWindowSizes(window_sizes, 0);
+    return window_sizes;
+  }
+
+  void VerifyWindowSizes(const std::array<std::array<int, axes>, axes> &window_sizes,
```

Comment:
```suggestion
  void ValidateWindowSizes(const std::array<std::array<int, axes>, axes> &window_sizes,
```
Verify - check if sth. is what it claims to be / should be / ... - e.g. you can verify credentials
Validate - check if sth. is correct.

Reply:
Good point, changed that.